src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/MaxNumberAssertsRuleTests.cs

```diff
@@ -0,0 +1,122 @@
+        {
+            AssertLinterRuleDiagnostics(MaxNumberAssertsRule.Code, text, diags =>
+            {
+                if (unusedParams.Any())
+                {
+                    diags.Should().HaveCount(unusedParams.Count());
+
+                    var rule = new MaxNumberAssertsRule();
+                    string[] expectedMessages = unusedParams.Select(p => rule.GetMessage(MaxNumberAssertsRule.MaxNumber)).ToArray();
+                    diags.Select(e => e.Message).Should().Contain(expectedMessages);
+                }
+                else
+                {
+                    diags.Should().BeEmpty();
+                }
+            },
+            new Options(OnCompileErrors.Ignore));
+        }
+
+        [DataRow(@"
+            assert a1 = true
+            assert a2 = true
+            assert a3 = true
+            assert a4 = true
+            assert a5 = true
+            assert a6 = true
+            assert a7 = true
+            assert a8 = true
+            assert a9 = true
+            assert a10 = true
+            assert a11 = true
+            assert a12 = true
+            assert a13 = true
+            assert a14 = true
+            assert a15 = true
+            assert a16 = true
+            assert a17 = true
+            assert a18 = true
+            assert a19 = true
+            assert a20 = true
+            assert a21 = true
+            assert a22 = true
+            assert a23 = true
+            assert a24 = true
+            assert a25 = true
+            assert a26 = true
+            assert a27 = true
+            assert a28 = true
+            assert a29 = true
+            assert a30 = true
+            assert a31 = true
+            assert a32 = true
+        ")]
+
+        [DataRow(@"
+            assert a1 = true
+            assert a2 = true
+            assert a3 = true
+            assert a4 = true
+            assert a5 = true
+            assert a6 = true
+            assert a7 = true
+            assert a8 = true
+            assert a9 = true
+            assert a10 = true
+            assert a11 = true
+            assert a12 = true
+            assert a13 = true
+            assert a14 = true
+            assert a15 = true
+            assert a16 = true
+            assert a17 = true
+            assert a18 = true
+            assert a19 = true
+            assert a20 = true
+            assert a21 = true
+            assert a22 = true
+            assert a23 = true
+            assert a24 = true
+            assert a25 = true
+            assert a26 = true
+            assert a27 = true
+            assert a28 = true
+            assert a29 = true
+            assert a30 = true
+            assert a31 = true
+            assert a32 = true
+            assert a33 = true
+            ",
+            "a1"
+        )]
+
+        [DataTestMethod]
+        public void TestRule(string text, params string[] unusedParams)
```

Comment:
```suggestion
        public void TestRule(string text, bool tooManyAsserts)
```

The name of the second parameter isn't right (this may have been copied unchanged from `NoUnusedParametersRuleTests` class?). The `CompileAndTest` method is checking for at most one "too many asserts" error message. so [a parameter array](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/params) isn't necessary; a simple boolean flag would be enough data for `CompileAndTest`

Reply:
I saw some similar tests like MaxNumberResourcesRuleTest and MaxNumberOutputsRuleTest also follow this pattern. Was there possibly a reason why these tests use an unused parameter to signal a violation of the rule?

Reply:
I'm guessing they were copied and pasted from another test (probably with `NoUnusedParametersRuleTests` being the original source). All of the `MaxNumber_Rule` linter rules can raise at most one diagnostic, and none of them are checking for unused `param` symbols.